### PR TITLE
Adding detail fatal error from backend to log output

### PR DESCRIPTION
**Pull request by mason-chase**
I'm very new to PowerDNS and I had to spend days to find out how PowerDNS works and why my YAML sample files taken from PowerDNS documentation didn't work.
- Please note that you have combined multiple scenario into one files at : https://doc.powerdns.com/md/authoritative/backend-geoip/ 
  and If e.what() was in logs , I would have known where to look straight.

### CODE REVIEW

**Comment by Habbie** [comment]
@cmouse want to have a look at this?

**Comment by cmouse** [comment]
Can you please do patch cleanup before review?

File: modules/geoipbackend/geoipbackend.cc [hunk 1]
```diff
@@ -124,172 +125,240 @@ void GeoIPBackend::initialize() {
...
-        rr.qname = qname;
-        if (rec->first.IsNull()) { 
-          rr.qtype = QType(0);
-        } else {
-          string qtype = boost::to_upper_copy(rec->first.as<string>());
-          rr.qtype = qtype;
+  if ( !getArg("zones-dir").empty() ) {
+    std::vector<std::string> extraZones;
+    gatherZones(extraZones); 
+    L << Logger::Info << "[GEOIPBACKEND][Parsing Zone Directory]: "<< getArg("zones-dir") << std::endl;
```
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: Inconsistent log format style, better formatted as

[geoipbackend]: Error while parsing zone directory: ...

File: modules/geoipbackend/geoipbackend.cc [hunk 2]
```diff
@@ -124,172 +125,240 @@ void GeoIPBackend::initialize() {
...
+    for(std::string& ZoneFile :  extraZones) {
+        if (!ZoneFile.c_str()) {
```
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: inconsistent variable naming style, better use zoneFile.

File: modules/geoipbackend/geoipbackend.cc [hunk 3]
```diff
@@ -124,172 +125,240 @@ void GeoIPBackend::initialize() {
...
+    }
+
+    if (!(dir = opendir(getArg("zones-dir").c_str()))) {
+       L << Logger::Error << getArg("zones-dir") << " is not accessible" << std::endl;
+       throw PDNSException(getArg("zones-dir") + " is not accessible");
+    }
+
+    while((ent = readdir(dir)) != NULL) {
+      if (ent->d_name[0] == '.') continue; // skip any dots
+      if (boost::ends_with(ent->d_name, ".yml")) {
```
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: these two could be combined as single if... `if (ent->d_name[0] != '.' && boost::ends_with(ent->d_name, ".yml") {`

File: modules/geoipbackend/geoipbackend.cc [hunk 4]
```diff
@@ -124,172 +125,240 @@ void GeoIPBackend::initialize() {
...
+        // build name
+        std::ostringstream namebuf;
+        namebuf << getArg("zones-dir").c_str() << "/" << ent->d_name; // FIXME: Use some path separator
```
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: maybe use boost::format instead or getArg("zones-dir") + "/" + ent->d_name? c_str() is not required here.

File: modules/geoipbackend/geoipbackend.cc [hunk 5]
```diff
@@ -124,172 +125,240 @@ void GeoIPBackend::initialize() {
...
 
-    // rectify the zone, first static records
-    for(auto &item : dom.records) {
-      // ensure we have parent in records
-      DNSName name = item.first;
-      while(name.chopOff() && name.isPartOf(dom.domain)) {
-        if (dom.records.find(name) == dom.records.end() && !dom.services.count(name)) { // don't ENT out a service!
+void GeoIPBackend::loadGeoFile(std::string &FilePath, vector<GeoIPDomain> &tmp_domains){
+    YAML::Node config;
+    config = YAML::LoadFile(FilePath);
```
Comment (cmouse) [inline modules/geoipbackend/geoipbackend.cc]: consistency, filePath

**Comment by cmouse** [comment]
I didn't do it fully yet, but you should also ensure it works when no zones-file is provided at all. Also, as general note, one should be aware that you cannot split configuration to multiple files for a zone, which should probably be documented. Documentation update is missing also.

**Comment by pieterlexis** [comment]
Closing for lack of response, please reply when updated to reopen

**Pull request closed without merging**